admin/services/generate-pins.service.js

```diff
@@ -63,14 +79,48 @@ const generatePinService = {
     return school
   },
 
+  /**
+   * Validate pin
+   * @param pin
+   * @param pinExpiresAt
+   * @returns {Boolean}
+   */
   isValidPin: (pin, pinExpiresAt) => {
     if (!pinExpiresAt || !pin) return false
     return moment(pinExpiresAt).isAfter(moment.utc())
   },
 
+  /**
+   * Generate Random Pin
+   * @param length
+   * @returns {String}
+   */
   generateRandomPin: (length) => {
     const chars = '23456789bcdfghjkmnpqrstvwxyz'
     return randomGenerator.getRandom(length, chars)
+  },
+
+  /**
+   * Get active school Password
+   * @param school
+   * @returns {String}
+   */
+  getActiveSchool: async (schoolId) => {
+    const school = await schoolDataService.findOne({_id: schoolId})
+    if (generatePinService.isValidPin(school.schoolPin, school.pinExpiresAt)) {
+      return school
+    } else return null
```

Comment:
It would be nicer to re-work this so that 

1. the school was returned at the bottom of the method
2. the else part was removed.

Reply:
@jon-shipley updated

Reply:
@GeorgeChatzigiannis - doesn't look updated